frontend/src/views/Abfrage.vue

```diff
@@ -644,361 +578,386 @@ export default class Abfrage extends Mixins(
+          abfragevariante.bauabschnitte[0].baugebiete = [technicalBaugebiet];
+          this.formChanged();
+        }
+      });
     }
-    return selectedBaugebiet;
   }
 
-  private getSelectedBaurate(abfrageTreeItem: AbfrageTreeItem): BaurateDto {
-    let selectedBaurate = undefined;
-    let selectedBaugebiet = this.getSelectedBaugebiet(abfrageTreeItem);
-    selectedBaurate = selectedBaugebiet.bauraten.find((baurate) => baurate === abfrageTreeItem.baurate);
-    if (_.isNil(selectedBaurate)) {
-      selectedBaurate = new BaurateModel(createBaurateDto());
+  private handleDetermineBauratenForBaugebiet(item: AbfrageTreeItem): void {
+    const baugebiet = item.value;
+
+    if (this.isBaugebiet(item, baugebiet)) {
+      this.determineBauraten(
+        baugebiet.realisierungVon,
+        baugebiet.gesamtanzahlWe,
+        baugebiet.geschossflaecheWohnen,
+        true
+      ).then((bauraten: Array<BaurateDto>) => {
+        baugebiet.bauraten = bauraten.map((baurate: BaurateDto) => new BaurateModel(baurate));
+        this.formChanged();
+      });
     }
-    return selectedBaurate;
   }
 
-  private handleSelectBauabschnitt(abfrageTreeItem: AbfrageTreeItem): void {
-    this.setAnzeigeContextAbfragevariante(abfrageTreeItem.contextAnzeigeAbfragevariante);
-    this.initializeFormulare();
-    this.selectedBauabschnitt = this.getSelectedBauabschnitt(abfrageTreeItem);
-    this.$nextTick(() => {
-      this.openBauabschnittFormular();
-    });
-  }
+  /* Diese "Ancestor"-Methoden sind prinzipiell dafür da, einen Vorfahren von einem bestimmten Typ zu finden.
+   * Jedoch können sie auch das übergebene Item selbst zurückgeben, wenn es dem Typen entspricht.
+   */
 
-  private handleDeleteBauabschnitt(abfrageTreeItem: AbfrageTreeItem): void {
-    this.setAnzeigeContextAbfragevariante(abfrageTreeItem.contextAnzeigeAbfragevariante);
-    this.bauabschnittTreeItemToDelete = abfrageTreeItem;
-    this.selectedBauabschnitt = this.getSelectedBauabschnitt(abfrageTreeItem);
-    this.openBauabschnittFormular();
-    this.isDeleteDialogBauabschnittOpen = true;
+  private getAncestorAbfragevariante(item: AbfrageTreeItem): AbfragevarianteModel | undefined {
+    while (item.parent) {
+      if (this.isAbfragevariante(item, item.value)) {
+        return item.value;
+      }
+      item = item.parent;
+    }
+
+    return undefined;
   }
 
-  private handleCreateNewBauabschnitt(abfrageTreeItem: AbfrageTreeItem): void {
-    this.setAnzeigeContextAbfragevariante(abfrageTreeItem.contextAnzeigeAbfragevariante);
-    let selectedAbfragevariante = this.getSelectedAbfragevariante(abfrageTreeItem);
-    this.selectedBauabschnitt = new BauabschnittModel(createBauabschnittDto());
-    if (_.isNil(selectedAbfragevariante.bauabschnitte)) {
-      selectedAbfragevariante.bauabschnitte = [];
+  private getAncestorBauabschnitt(item: AbfrageTreeItem): BauabschnittModel | undefined {
```

Comment:
Ist die Benennung `getAncestorBauabschnitt` korrekt? Der Methodenname deutet darauf hin, dass entweder das Elternobjekt eines Bauabschnitts zurückgegeben wird oder falls das Objekt im Parameter ein Baugebiet ist, der Bauabschnitt zurückgegeben wird.

Tatsächlich wird jedoch im Falle einer Abfragevariaten ein technischer Bauabschnitt erstellt, im Falle einer Baugebiets wird der Elternbauabschitt zurückgegeben und im Fall einer Baurate für das Elternbaugebiet der Bauabschnitt ermittelt

![image](https://github.com/it-at-m/isi-frontend/assets/56300801/6f8dc2d5-fdeb-46a5-b449-cf8129108b8d)

Reply:
Mit `getAncestorBauabschnitt` war eher "hole mir den ersten Vorfahren dieser Entität, welcher ein Bauabschnitt ist" gemeint. Hast du Vorschläge, wie ichs klarer benennen könnte?

Reply:
Evtl. etwas genauer im Methodenamen die Logik der Methode hinterlegen.

z.B. `getNearestAncestorOfTypeBauabschnittForGivenTreeItem`

Reply:
Benennung angepasst.